### PR TITLE
[Merged by Bors] - feat: port algebra.group.semiconj

### PR DESCRIPTION
**Pull request by dwarn**
mathlib SHA e50b8c261b0a000b806ec0e1356b41945eda61f7

This completes an earlier partial port of the file.

~~I added a workaround for the issue explained by Floris at https://github.com/leanprover-community/mathlib4/pull/707#issuecomment-1326669750.~~

### CODE REVIEW

**Comment by urkud** [comment]
I see that the `mathlib3port` file has a lot of "dubious translation" warnings. What do they mean?

File: Mathlib/Algebra/Group/Semiconj.lean [hunk 1]
```diff
@@ -91,5 +153,69 @@ theorem pow_right {a x y : M} (h : SemiconjBy a x y) (n : ℕ) : SemiconjBy a (x
...
+@[simp, to_additive addSemiconjBy_iff_eq]
+theorem semiconjBy_iff_eq [CancelCommMonoid M] {a x y : M} : SemiconjBy a x y ↔ x = y :=
+  ⟨fun h => mul_left_cancel (h.trans (mul_comm _ _)), fun h => by rw [h, SemiconjBy, mul_comm]⟩
+#align semiconj_by_iff_eq semiconjBy_iff_eq
+
+/-- `a` semiconjugates `x` to `a * x * a⁻¹`. -/
+@[to_additive "`a` semiconjugates `x` to `a + x + -a`."]
+theorem Units.mk_semiconj_by [Monoid M] (u : Mˣ) (x : M) : SemiconjBy (↑u) x (u * x * ↑u⁻¹) := by
+  unfold SemiconjBy; rw [Units.inv_mul_cancel_right]
+#align units.mk_semiconj_by Units.mk_semiconj_by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Group/Semiconj.lean:221]:
```suggestion
theorem Units.mk_semiconjBy [Monoid M] (u : Mˣ) (x : M) : SemiconjBy (↑u) x (u * x * ↑u⁻¹) := by
  unfold SemiconjBy; rw [Units.inv_mul_cancel_right]
#align units.mk_semiconj_by Units.mk_semiconjBy
```

File: Mathlib/Algebra/Group/Semiconj.lean [hunk 2]
```diff
@@ -26,26 +26,23 @@ This file provides only basic operations (`mul_left`, `mul_right`, `inv_right` e
...
-def SemiconjBy {M : Type u} [Mul M] (a x y : M) : Prop :=
+def SemiconjBy [Mul M] (a x y : M) : Prop :=
   a * x = y * a
+#align semiconj_by SemiconjBy
 
 namespace SemiconjBy
 
-/-- Equality behind `semiconj_by a x y`; useful for rewriting. -/
+/-- Equality behind `SemiconjBy a x y`; useful for rewriting. -/
 @[to_additive "Equality behind `add_semiconj_by a x y`; useful for rewriting."]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Group/Semiconj.lean:38]:
```suggestion
@[to_additive "Equality behind `AddSemiconjBy a x y`; useful for rewriting."]
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

**Comment by semorrison** [comment]
You seem to not be authorized

**Comment by Ruben-VandeVelde** [comment]
Oh, I may have misunderstood how that worked. Anyway, this one seems ready to go

**Comment by semorrison** [comment]
@Ruben-VandeVelde, sorry, the maintainer merge bot was misconfigured. Hopefully it will work now.

File: Mathlib/Algebra/Group/Semiconj.lean [hunk 3]
```diff
@@ -56,33 +53,98 @@ theorem mul_right (h : SemiconjBy a x y) (h' : SemiconjBy a x' y') :
...
-/-- If both `a` and `b` semiconjugate `x` to `y`, then so does `a * b`. -/
-@[to_additive "If both `a` and `b` semiconjugate `x` to `y`, then so does `a + b`."]
+/-- If `b` semiconjugates `x` to `y` and `a` semiconjugates `y` to `z`, then `a * b`
+semiconjugates `x` to `z`. -/
+@[to_additive "If `b` semiconjugates `x` to `y` and `a` semiconjugates `y` to `z`, then `a + b`
+semiconjugates `x` to `z`."]
 theorem mul_left (ha : SemiconjBy a y z) (hb : SemiconjBy b x y) : SemiconjBy (a * b) x z := by
   unfold SemiconjBy
   rw [mul_assoc, hb.eq, ←mul_assoc, ha.eq, mul_assoc]
+#align semiconj_by.mul_left SemiconjBy.mul_left
```
Comment (semorrison) [inline Mathlib/Algebra/Group/Semiconj.lean:65]: There should be a separate #align for the additive version. I guess we need a linter for this.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**